### PR TITLE
Detecting that build should be a release when:

### PR DESCRIPTION
**Pull request by bgalek**
* branch or tag that triggered the workflow (GITHUB_REF) is valid semver
* current commit is tagged with tag that is valid semver

### CODE REVIEW

File: maven/index.js [hunk 1]
```diff
@@ -16,12 +18,25 @@ const version = core.getInput('version');
...
+}
+
+function validSemVer(tag) {
+  return semver.valid(stripReleaseTagPrefix(tag));
+}
+
+const currentTagNames = exec('git tag --points-at=HEAD').toString().split('\n');
 core.info(`current branch: ${currentBranch}`);
-const isSnapshot = !['master', 'main'].includes(currentBranch);
+core.info(`current tag names: ${currentTagNames}`);
```
Comment (bgalek) [inline maven/index.js:36]: .join(',');

File: maven/index.js [hunk 2]
```diff
@@ -16,12 +18,25 @@ const version = core.getInput('version');
 const tychoPath = core.getInput('tycho');
 const provisioningPath = core.getInput('provisioning');
 const distributionsDir = core.getInput('distributionsDir');
+const releaseTagPrefix = core.getInput('releaseTagPrefix');
 const currentBranch = getBranchName();
 
 reportAction();
 
+function stripReleaseTagPrefix(tag) {
```
Comment (bgalek) [inline maven/index.js:26]: move declarations to bottom of the file

File: maven/index.js [hunk 3]
```diff
@@ -16,12 +18,25 @@ const version = core.getInput('version');
...
+  return semver.valid(stripReleaseTagPrefix(tag));
+}
+
+const currentTagNames = exec('git tag --points-at=HEAD').toString().split('\n');
 core.info(`current branch: ${currentBranch}`);
-const isSnapshot = !['master', 'main'].includes(currentBranch);
+core.info(`current tag names: ${currentTagNames}`);
+core.info(`releaseTagPrefix: ${releaseTagPrefix}`);
+
+const isSnapshot = !['master', 'main'].includes(currentBranch) && !validSemVer(currentBranch) && !currentTagNames.some( tag => validSemVer(tag));
```
Comment (bgalek) [inline maven/index.js:39]: `some( t` -> lint `some(t`

**Review by bgalek — COMMENTED** [review]
please update readme :)

**Review by pawel-piecyk — APPROVED** [review]

**Pull request merged**